src/ValidationContext.php

```diff
@@ -6,42 +6,68 @@
 
 use RuntimeException;
 use Yiisoft\Arrays\ArrayHelper;
+use Yiisoft\Validator\Rule\Each;
+use Yiisoft\Validator\Rule\Nested;
 
 /**
  * Validation context that might be taken into account when performing validation.
+ * The context is passed to both direct calls of {@see ValidatorInterface::validate()} and when {@see Nested}
+ * or {@see Each} is used.
  *
  * @psalm-import-type RulesType from ValidatorInterface
  */
 final class ValidationContext
 {
+    /**
+     * @var ValidatorInterface|null A validator instance. Null means context data was not set yet
+     * with {@see setContextDataOnce()}.
+     */
     private ?ValidatorInterface $validator = null;
 
     /**
-     * @var mixed The raw validated data.
+     * @var mixed The raw validated data. Null means context data was not set yet with {@see setContextDataOnce()}.
      */
     private mixed $rawData = null;
 
     /**
-     * @var DataSetInterface|null Data set the attribute belongs to. Null if data set not set.
+     * @var DataSetInterface|null Data set the attribute belongs to.
+     * Null if data set was not set yet with {@see setDataSet()}.
      */
     private ?DataSetInterface $dataSet = null;
 
     /**
-     * @var string|null Validated attribute name. Null if a single value is validated.
+     * @var string|null Validated data set attribute name. Null if a single value is validated.
      */
     private ?string $attribute = null;
 
+    /**
+     * @var AttributeTranslatorInterface|null Default attribute translator to use if attribute translator is not set.
+     */
     private ?AttributeTranslatorInterface $defaultAttributeTranslator = null;
 
     /**
      * @param array $parameters Arbitrary parameters.
```

Comment:
Suggest to use "var" and place right above corresponding arguments. Because they are also class attributes and for consistency.